nidaqmx_examples/Analog_Generation/Current/Cont_Gen_Cur_Wfm-Int_Clk.py

```diff
@@ -0,0 +1,267 @@
+  """
+  This function helps to set the Sample Rate for the On-Demand task, based on the samples per buffer,
+  Cycles per buffer and desired Waveform Frequency. It also sets the Acquisition Mode (Finite / Continuous)
+  and the samples per channel.
+
+  Inputs:
+    desiredFreq : Is the desired Frequency for the selected Waveform [Hz]
+    sampsPerBuff: Number of samples to write in the output buffer
+    cyclesPerBuff: Number of cycles within a buffer write operation
+    sampsPerCycle: Number of samples to write on a single cycle
+
+  Outputs:
+    desiredSampClkRate: Is the calculated sample rate to be used at the DAQmx timing configuration.
+    resultingFrequency: I DON'T KNOW WHAT THIS WAS USED FOR 
+  """
+  sampsPerCycle = sampsPerbuff / cyclesPerBuff
+  desiredSampClkRate = desiredFreq * sampsPerbuff / cyclesPerBuff
+  resultingFrequency = desiredSampClkRate / sampsPerCycle
+
+  try:
+    DAQmxTask.timing.cfg_samp_clk_timing(desiredSampClkRate, sample_mode = AcquisitionType.CONTINUOUS)
+
+  except DAQmxError.DAQError as err:
+    print("DAQmx error", err.error_code, "occurred")
+    print("Cause: ", err.error_type)
+    exit_application()
+
+#----------------------
+# Waveform Generation functions
+#---------------------
+def GenSineWave(numElements, amplitude, frequency, phase):
+  print ("Using Sine Wave")
+
+  for i in range(numElements):
+    data.append(offset + (amplitude * math.sin((math.pi/180)*(phase + 360.0 * frequency * i))))
+    phase = math.fmod(phase + frequency * 360 * numElements, 360)
+
+
+def GenSquareWave(numElements, amplitude, frequency, phase, dutyCycle):
+  print ("Using Square Wave")
+
+  for i in range(numElements):
+    phase_i = math.fmod(phase + 360* frequency *i, 360)
+    if phase_i/360 <= dutyCycle/100:
+      data.append(offset + amplitude)
+    else : 
+      data.append(offset -amplitude)
+
+def GenSawtoothWave(numElements, amplitude, frequency, phase):
+  print ("Using Saw tooth Wave")
+
+  for i in range(numElements):
+    phase_i = math.fmod(phase + 360* frequency *i, 360)
+    percentPeriod = phase_i/360
+    dat = amplitude* 2.0 * percentPeriod
+
+    if percentPeriod <= 0.5:
+          data.append(offset + dat)
+    else:
+      data.append(offset + (dat -2.0 * amplitude))
+
+def GenTriangleWave(numElements, amplitude, frequency, phase):
+  """
+  This function is used to crate a Triangle Waveform to be used as output
+  in a DAQmx task.
+
+  numElements:
+  amplitude:
+  frequency:
+  phase: 
+  """
+  print ("Using Triangle Wave")
+
+  for i in range(numElements):
+      phase_i = math.fmod(phase + 360*frequency*i, 360)
+      percentPeriod = phase_i/360
+      dat = amplitude* 4.0 * percentPeriod
+
+      if percentPeriod <= 0.25:
+        data.append(offset + dat)
+      elif percentPeriod <= 0.75:
+        data.append(offset + (2.0 * amplitude - dat))
+
+      else:
+        data.append(offset + (dat - 4.0 * amplitude))
+
+  phase = math.fmod(phase + frequency * 360 * numElements, 360)
```

Comment:
nothing uses phase after this... what is this doing? was it meant to be in the loop?